spec/models/user_spec.rb

```diff
@@ -243,33 +263,46 @@
     end
   end
 
-  describe 'groups' do
+  describe '#groups' do
+    subject { user.groups }
+    let(:user) { User.find_or_create_by_webauth(double('webauth', :login => 'asdf', :logged_in? => true, :privgroup => webauth_groups.join('|'))) }
+
     context 'specified' do
-      before :each do
-        @webauth_privgroup_str = 'dlss:testgroup1|dlss:testgroup2|dlss:testgroup3'
-        @user = User.find_or_create_by_webauth(double('webauth', :login => 'asdf', :logged_in? => true, :privgroup => @webauth_privgroup_str))
+      let(:webauth_groups) { %w(dlss:testgroup1 dlss:testgroup2 dlss:testgroup3) }
+
+      it 'returns the groups by webauth' do
+        expected_groups = ['sunetid:asdf'] + webauth_groups.map { |g| "workgroup:#{g}" }
+        expect(subject).to eq(expected_groups)
       end
-      it 'should return the groups by webauth' do
-        expected_groups = ['sunetid:asdf'] + @webauth_privgroup_str.split(/\|/).map { |g| "workgroup:#{g}" }
-        expect(@user.groups).to eq(expected_groups)
+    end
+
+    context 'when impersonating' do
+      before do
+        user.set_groups_to_impersonate(%w(workgroup:dlss:impersonatedgroup1 workgroup:dlss:impersonatedgroup2))
       end
-      it 'should return the groups by impersonation' do
-        impersonated_groups = %w(workgroup:dlss:impersonatedgroup1 workgroup:dlss:impersonatedgroup2)
-        @user.set_groups_to_impersonate(impersonated_groups)
-        expect(@user.groups).to eq(impersonated_groups)
```

Comment:
i like the more thorough tests for what shouldn't come through from the actual webauth groups, but why lose the simple test for whether `#groups` returns the impersonated group list?  e.g., why not `let(:groups_to_impersonate) { %w(workgroup:dlss:impersonatedgroup1 workgroup:dlss:impersonatedgroup2) }`, use that in the `user.set_groups_to_impersonate` call, and then have something like
```ruby
it 'returns the impersonated groups' do
  expect(subject).to eq groups_to_impersonate
end
```